com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -366,53 +348,112 @@ internal Vector3 Scale
+
+        internal NetworkVariable<NetworkTransformState> ReplicatedNetworkState
+        {
+            get
+            {
+                if (!IsServerAuthoritative())
+                {
+                    return m_ReplicatedNetworkStateOwner;
+                }
+
+                return m_ReplicatedNetworkStateServer;
+            }
+        }
+
+        private NetworkTransformState m_LocalAuthoritativeNetworkState;
 
         private bool m_HasSentLastValue = false; // used to send one last value, so clients can make the difference between lost replication data (clients extrapolate) and no more data to send.
 
 
-        private BufferedLinearInterpolator<float> m_PositionXInterpolator; // = new BufferedLinearInterpolatorFloat();
-        private BufferedLinearInterpolator<float> m_PositionYInterpolator; // = new BufferedLinearInterpolatorFloat();
-        private BufferedLinearInterpolator<float> m_PositionZInterpolator; // = new BufferedLinearInterpolatorFloat();
-        private BufferedLinearInterpolator<Quaternion> m_RotationInterpolator; // = new BufferedLinearInterpolatorQuaternion(); // rotation is a single Quaternion since each euler axis will affect the quaternion's final value
-        private BufferedLinearInterpolator<float> m_ScaleXInterpolator; // = new BufferedLinearInterpolatorFloat();
-        private BufferedLinearInterpolator<float> m_ScaleYInterpolator; // = new BufferedLinearInterpolatorFloat();
-        private BufferedLinearInterpolator<float> m_ScaleZInterpolator; // = new BufferedLinearInterpolatorFloat();
+        private BufferedLinearInterpolator<float> m_PositionXInterpolator;
+        private BufferedLinearInterpolator<float> m_PositionYInterpolator;
+        private BufferedLinearInterpolator<float> m_PositionZInterpolator;
+        private BufferedLinearInterpolator<Quaternion> m_RotationInterpolator;
+        private BufferedLinearInterpolator<float> m_ScaleXInterpolator;
+        private BufferedLinearInterpolator<float> m_ScaleYInterpolator;
+        private BufferedLinearInterpolator<float> m_ScaleZInterpolator;
         private readonly List<BufferedLinearInterpolator<float>> m_AllFloatInterpolators = new List<BufferedLinearInterpolator<float>>(6);
 
         private Transform m_Transform; // cache the transform component to reduce unnecessary bounce between managed and native
         private int m_LastSentTick;
         private NetworkTransformState m_LastSentState;
 
+        internal NetworkTransformState GetLastSentState()
+        {
+            return m_LastSentState;
+        }
+
         /// <summary>
-        /// Tries updating the server authoritative transform, only if allowed.
-        /// If this called server side, this will commit directly.
-        /// If no update is needed, nothing will be sent. This method should still be called every update, it'll self manage when it should and shouldn't send
+        /// Refer to <see cref="TryCommitTransform">, this is a legacy method.
```

Comment:
Could you use the Obsolete attribute?
https://docs.microsoft.com/en-us/dotnet/api/system.obsoleteattribute?view=net-6.0

Reply:
The Obsolete tag is a breaking change as it causes either a compile error, or a warning, for anyone using the method. That means introducing it requires a Major version revision. It's effectively useless in the Unity ecosystem unless it's paired with an ApiUpdater, unfortunately.